projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -464,13 +467,177 @@ async def run(self) -> int:
             await register_vector_async(conn)
             while True:
                 if not self._continue_processing(loops, res):
+                    await self._check_and_process_openai_batches(conn)
+                    # TODO how can we run this only after hitting the rate limit of the normal openai batch embedding api?
```

Comment:
I am not entirely sure this is how it should work.
My gut idea would be to have a flag on the `create_vectorizer` call that allows enabling and disabling the batch_api. I'd then also put a configurable threshhold there, which would trigger the batch calls when the queue grows e.g. larger than 1000 items?

Not entirely sure about this interface. Maybe @alejandrodnm has another idea.

Reply:
I think making this configurable makes sense. It will be easier to implement at least :grin: 

How would I create a new config variable for this? I know how to add a new env variable for this, but maybe that's not what we want?